src/main/java/org/opensearch/geospatial/ip2geo/cache/GeoDataCache.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo.cache;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+
+import org.opensearch.common.cache.Cache;
+import org.opensearch.common.cache.CacheBuilder;
```

Comment:
why we are going with an internal [OpenSearch cache implementation](https://github.com/opensearch-project/OpenSearch/blob/0ca51a774211184835c4825dfeff38b23198352e/server/src/main/java/org/opensearch/common/cache/Cache.java#L54-L84), when there are other tested and more prominent libraries for cache like Google Guava etc. Example: https://github.com/opensearch-project/k-NN/blob/8e2ad4595bd4f7c64d0b312cac30de8a42e4bf0b/src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java#L15-L14

Can we do some deep-dive here to figure out which one to choose and why?

Reply:
Current GeoIP processor use OpenSearch cache implementation which is well tested. Why do we want to explore new option here when there is working option already by adding additional dependency on third party library?